File: c#/1726. Tuple with Same Product.cs
public class Solution {
    public int TupleSameProduct(int[] nums) {
        Dictionary<int, int> map = new();
        int n = nums.Length;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int res = nums[i] * nums[j];
                if (!map.ContainsKey(res)) map[res] = 0;
                map[res]++;
            }
        }

        return map.Values.Sum(x => x * (x - 1) * 4);
    }
}

File: javascript/1726. Tuple with Same Product.js
var tupleSameProduct = function (nums) {
  let n = nums.length;
  let map = new Map();
  for (let i = 0; i < n; i++) {
    for (let j = i + 1; j < n; j++) {
      let res = nums[i] * nums[j];
      map.set(res, (map.get(res) ?? 0) + 1);
    }
  }

  let ans = 0;
  for (let [, val] of map) {
    ans += val * (val - 1) * 4;
  }

  return ans;
};
